From a46ae40296176d9379d61c9653c92a621deef328 Mon Sep 17 00:00:00 2001
From: "Slash.Huang" <slash.huang@regulus.com>
Date: Mon, 26 Sep 2022 13:47:50 +0800
Subject: [PATCH 06/25] light-sensor: isl76683: fixed rcu issue with sdhci

---
 drivers/iio/light/isl76683.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/iio/light/isl76683.c b/drivers/iio/light/isl76683.c
index 034a009b3de7..d99b4f23851e 100644
--- a/drivers/iio/light/isl76683.c
+++ b/drivers/iio/light/isl76683.c
@@ -502,10 +502,10 @@ static int isl76683_probe(struct i2c_client *client)
 	chip->trig->dev.parent = dev;
 	iio_trigger_set_drvdata(chip->trig, chip);
 
-	ret = devm_request_irq(dev, client->irq,
+	ret = devm_request_threaded_irq(dev, client->irq, NULL,
 			isl76683_interrupt_handler,
-			IRQF_TRIGGER_FALLING | IRQF_ONESHOT,
-			"isl76683_event", indio_dev);
+			IRQ_TYPE_EDGE_FALLING | IRQF_ONESHOT, "isl76683_event",
+			indio_dev);
 	if (ret) {
 		dev_err(dev, "irq request error\n");
 		return ret;
-- 
2.25.1

