From 5a2f8c2b0eb03ae9996b7ab4879d37ac2943f0e7 Mon Sep 17 00:00:00 2001
From: slash <slash@gmail.com>
Date: Mon, 7 Nov 2022 13:30:08 +0800
Subject: [PATCH 18/18] [gnk-g2l] dts: enable scif1/scif3 and dsi

---
 .../dts/renesas/r9a07g044l2-smarc-gnk.dts     | 45 +++++++++++++++++++
 1 file changed, 45 insertions(+)

diff --git a/arch/arm64/boot/dts/renesas/r9a07g044l2-smarc-gnk.dts b/arch/arm64/boot/dts/renesas/r9a07g044l2-smarc-gnk.dts
index 9449872f1..624245c98 100644
--- a/arch/arm64/boot/dts/renesas/r9a07g044l2-smarc-gnk.dts
+++ b/arch/arm64/boot/dts/renesas/r9a07g044l2-smarc-gnk.dts
@@ -11,7 +11,52 @@
 #include "rzg2l-smarc-pinfunction.dtsi"
 #include "rz-gnk-smarc-common.dtsi"
 
+
 / {
 	model = "Renesas Gnk Board based on r9a07g044l2";
 	compatible = "renesas,smarc-gnk", "renesas,r9a07g044l2", "renesas,r9a07g044";
 };
+
+&scif1 {
+	pinctrl-0 = <&scif1_pins>;
+	pinctrl-names = "default";
+	status = "okay";
+};
+
+&scif3 {
+	pinctrl-0 = <&scif3_pins>;
+	pinctrl-names = "default";
+	status = "okay";
+};
+
+
+&pinctrl {
+	/delete-node/ can0-stb;
+	/delete-node/ can1-stb;
+	scif1_pins: scif1 {
+		pinmux = <RZG2L_PORT_PINMUX(40, 0, 1)>,	/* TxD */
+			 <RZG2L_PORT_PINMUX(40, 1, 1)>;	/* RxD */
+	};
+
+	scif3_pins: scif3 {
+		pinmux = <RZG2L_PORT_PINMUX(0, 0, 5)>,	/* TxD */
+			 <RZG2L_PORT_PINMUX(0, 1, 5)>;	/* RxD */
+	};
+
+	pmod0-gpio {
+		gpio-hog;
+		pinmux = <RZG2L_GPIO(43, 0) GPIO_ACTIVE_HIGH>,
+			<RZG2L_GPIO(43, 2) GPIO_ACTIVE_HIGH>,
+			<RZG2L_GPIO(41, 0) GPIO_ACTIVE_HIGH>,
+			<RZG2L_GPIO(43, 1) GPIO_ACTIVE_HIGH>;
+		output-low;
+	};
+};
+
+&gpt4 {
+	status = "disabled";
+};
+
+&dsi0 {
+	status = "okay";
+};
-- 
2.25.1

