From ae4a36b685439a00fc91c21e59af752ef725cd22 Mon Sep 17 00:00:00 2001
From: slash <slash@gmail.com>
Date: Thu, 12 Jan 2023 09:13:11 +0800
Subject: [PATCH 18/25] lightsensor: isl76683 disable interrupt

---
 drivers/iio/light/isl76683.c | 11 +++--------
 1 file changed, 3 insertions(+), 8 deletions(-)

diff --git a/drivers/iio/light/isl76683.c b/drivers/iio/light/isl76683.c
index d99b4f23851e..18eae24777df 100644
--- a/drivers/iio/light/isl76683.c
+++ b/drivers/iio/light/isl76683.c
@@ -161,14 +161,7 @@ static int isl76683_singleshot_conversion(struct isl76683_chip *chip, int *val)
 	/* wait for measurement to complete */
 	timeout = wait_for_completion_interruptible_timeout(
 			&chip->irq_complete,
-			msecs_to_jiffies(5000));
-	if (timeout == 0) {
-		dev_err(&chip->client->dev, "measurement timed out\n");
-		return -ETIMEDOUT;
-	} else if (timeout < 0) {
-		dev_err(&chip->client->dev, "wait for measurement failed\n");
-		return -EINTR;
-	}
+			usecs_to_jiffies(5000));
 
 	ret = isl76683_get_sensordata(chip, val);
 	if (ret) {
@@ -347,6 +340,7 @@ static int isl76683_read_raw(struct iio_dev *indio_dev,
 			iio_device_release_direct_mode(indio_dev);
 			return ret;
 		}
+
 		ret = isl76683_singleshot_conversion(chip, val);
 		iio_device_release_direct_mode(indio_dev);
 		return ret;
@@ -510,6 +504,7 @@ static int isl76683_probe(struct i2c_client *client)
 		dev_err(dev, "irq request error\n");
 		return ret;
 	}
+	disable_irq(client->irq);
 
 	ret = devm_iio_trigger_register(dev, chip->trig);
 	if (ret) {
-- 
2.25.1

